server.js

```diff
@@ -98,20 +107,26 @@ function join_custom_room(ws, join_room_json) {
     // Add a prefix to the room id to indicate custom match.
     room_id = "custom_" + room_id
 
+    // Extract room settings from the passed in json.
     var deck_id = join_room_json.deck_id
+    var starting_timer = join_room_json.starting_timer
+    var enforce_timer = join_room_json.enforce_timer
     var player = active_connections.get(ws)
     player.set_deck_id(deck_id)
     var success = false
+
+    // Check whether the player is joining an existing room.
     if (game_rooms.hasOwnProperty(room_id)) {
       const room = game_rooms[room_id]
+      // Check for Player/Room version mismatch.
       if (room.version != version) {
-        // Player/Room version mismatch.
         send_join_version_error(ws)
         return true
       }
       success = room.join(player)
     } else {
-      const new_room = new GameRoom(version, room_id, database)
+      // Start a new custom game room.
+      const new_room = new GameRoom(version, room_id, database, starting_timer, enforce_timer)
```

Comment:
You need to update the other location that calls new GameRoom in this file in create_new_match_room().
Either pass in some default params or update GameRoom constructor to have default params.